### ISSUE TITLE
Why not just call `fold` `reduce`?

### ISSUE THREAD

**Issue by sartaj** [staltz/xstream#111]
If part of the goal of xstream is to simplify understanding of streams and make it easier for beginner js devs to understand, why not try to use native analogies when available?

For example, teaching a beginner coder reactive programming requires you to teach native map/reduce in arrays anyways, so the more overlap in concept there is here, the easier it is to each them. While `fold` may be in other apis, it's not native in js, so `fold` becomes very specific to xstream. Conceptually `reduce` is something that when the user learns will be valuable for a long time.

Just food for thought. Part of this is me trying to simplify the learning of stream libraries: https://github.com/sartaj/simply-reactive

**Comment by staltz** [comment]
Reduce on arrays returns **one value**, not an array.

Also, reduce in RxJS returns an Observable with **one value** at the end of the input Observable. That's the idea of reduce: take a collection and reduce it to one thing.

Fold is inspired by Elm's `foldp` as in "fold in time". I like the physical analogy of folding paper, it's very easy to teach and understand.

**Comment by sartaj** [comment]
Cool, I understand. 

I wonder if there is a way to distill all the analogies within xstream/cycle/rx. Right now there seems to be so many, when I teach them I've found students not able to reconcile one analogy with another. xstream does a great job removing a lot of these. I'm wondering if it can be taken further.

Here's some I've found (not all related to xstream, but in reactive programming):

* Water utilities (streams, pipes, backpressure, pool, sources, sinks)
* Radio systems (subscribe, publish, observable, signals)
* Grammar (subject, predicate)
* Sensory (observable, observe)
* Computer hardware related (drivers)
* Paper (fold)
* Mathmatical function related (map, flatten)
* And sheer mind games (wtf is a flatmap)
